content/docs/tasks.md

```diff
@@ -0,0 +1,178 @@
+      args:
+        - wget -O /tmp/index.html 'http://cowsay.morecode.org/say?message=Good+things+come+when+you+least+expect+them&format=html'
+      volumeMounts:
+        - name: shared-data
+          mountPath: /tmp
+```
+
+Here we use an online [cowsay-generator](http://cowsay.morecode.org) API to generate and download an `index.html` file and save in the mounted volume under `/tmp/index.html` path.  
+
+Given the above pipe-pod and `genwww` task specification, KUDO will run the pipe-pod, wait for the successful generation of the `index.html`, copy it out and save it as a ConfigMap. Now, this file is ready to be used in our Nginx deployment spec. Let's extend the `templates/deployment.yaml` to mount the generated file:
+
+```yaml
+...
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.7.9
+      ports:
+        - containerPort: 80
+      volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html/
+  volumes:
+    - name: www
+      configMap:
+        name: {{ .Pipes.indexHtml }}
+```
+
+We successfully created a [mount volume from the ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#populate-a-volume-with-data-stored-in-a-configmap). Note the parametrized ` {{ .Pipes.indexHtml } }` configMap name. `.Pipes` keyword allows us to address pipe-artifacts and `indexHtml` is the key we defined in the pipe-task above.
+
+Now that we've prepared our pipe-task and the corresponding resources we can use it as part of the deploy plan:
+
+```yaml
+...
+plans:
+  deploy:
+    strategy: serial
+    phases:
+      - name: main
+        strategy: serial
+        steps:
+          - name: genfiles
+            tasks:
+              - genwww
+          - name: app
+            tasks:
+              - app
+```
+
+**Note** that the `main` phase has a `serial` strategy. We need to wait for the `genfiles` step to finish successfully before pipe-artifact can be used in a subsequent step.
+
+Once our Nginx server is up and running we should be able to read a pearl of cow wisdom:
+
+```text
+  ______________________________________
+/ Good things come when you least expect \
+\ them                                   /
+  --------------------------------------
+         \   ^__^
+          \  (oo)\_______
+             (__)\       )\/\
+                 ||----w |
+                 ||     ||
+```  
+
+**Note:**
+
+- File generating Pod has to be side-effect free (meaning side-effects that are observable outside of the container like a 3rd party API call) as the container might be executed multiple times on failure. A `restartPolicy: OnFailure` is used for the pipe-pod
+- Only files <1Mb are applicable to be stored as ConfigMap or Secret. A pipe-task will fail should it try to copy files >1Mb
+- As of KUDO 0.9.0 pipe-artifacts can only be used within the same plan they were generated
+- You can pipeline artifacts by creating them in one pipe-task and mounting in a subsequent one
```

Comment:
hmm I am not sure what this means, can you elaborate? Oh you mean that you can have multiple pipe tasks each reading input from the previous one?

Reply:
> Oh you mean that you can have multiple pipe tasks each reading input from the previous one?

exactly. would you phrase it differently?